embed-code-go/embed_code.go

```diff
@@ -0,0 +1,177 @@
+//
+// docsRoot — a path to a root directory with docs files.
+//
+// configPath — a path to a yaml configuration file which contains the roots.
+//
+// checkUpToDate — true to check for code embeddings to be up-to-date. Otherwise, the embedding is performed.
+type flags struct {
+	codeRoot      string
+	docsRoot      string
+	configPath    string
+	checkUpToDate bool
+}
+
+// Struct with roots that contained in a yaml configuration file.
+//
+// codeRoot — a path to a root directory with code files.
+//
+// docsRoot — a path to a root directory with docs files.
+type configFields struct {
+	CodeRoot string `yaml:"code_root"`
+	DocsRoot string `yaml:"docs_root"`
+}
+
+// Reads the roots from the provided configPath and returns a configFields struct.
+//
+// configPath — a path to a yaml configuration file which contains the roots.
+//
+// Returns a configFields struct filled with the roots.
+func readRootsFromConfig(configPath string) configFields {
+	content, err := os.ReadFile(configPath)
+	if err != nil {
+		panic(err)
+	}
+
+	configFields := configFields{}
+	err = yaml.Unmarshal(content, &configFields)
+	if err != nil {
+		panic(err)
+	}
+
+	return configFields
+}
+
+// Reads user-specified flags from the command line.
+//
+// Returns a flags struct filled with the corresponding flags.
+func readFlags() flags {
+	codeRoot := flag.String("code_root", "", "a path to a root directory with code files")
+	docsRoot := flag.String("docs_root", "", "a path to a root directory with docs files")
+
+	configPath := flag.String("config_path", "",
+		"a path to a yaml configuration file, which contains 'code_root' and 'docs_root' fields")
+
+	checkUpToDate := flag.Bool("up_to_date", false,
+		"true to check for code embeddings to be up-to-date, false to perform embedding")
+
+	flag.Parse()
+
+	return flags{
+		codeRoot:      *codeRoot,
+		docsRoot:      *docsRoot,
+		configPath:    *configPath,
+		checkUpToDate: *checkUpToDate,
+	}
+
+}
+
+// Checks the validity of user-provided flags and returns an error message if any of the validation rules are broken.
+// If everything is ok, returns an empty string.
+//
+// flagsSet — a struct with user-provided flags.
+func validate(flagsSet flags) string {
+	isRootsSet := flagsSet.codeRoot != "" && flagsSet.docsRoot != ""
+	isOneOfRootsSet := flagsSet.codeRoot != "" || flagsSet.docsRoot != ""
+	isConfigSet := flagsSet.configPath != ""
+
+	validationMessage := ""
+
+	if isConfigSet && isOneOfRootsSet {
+		return "Config path cannot be set when code_root and docs_root are set."
+	}
+	if isOneOfRootsSet && !isRootsSet {
+		return "If one of code_root and docs_root is set, the another one must be set as well."
+	}
+	if !(isRootsSet || isConfigSet) {
+		return "Embed code should be used with either config_path or both code_root and docs_root being set."
+	}
+
+	return validationMessage
+}
+
+// Generates and returns a configuration based on the provided flags.
+//
+// flagsSet — a struct with user-provided flags.
```

Comment:
I have previously mentioned they are not flags. And in this case, it's for sure not a set of `flags`. Set of `flags` is a structure, in which there may be many instances of `flags`, and they aren't ordered (and most likely, they aren't repeated).